internal/product/accessor_test.go

```diff
@@ -0,0 +1,225 @@
+package product
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/onsi/gomega"
+)
+
+func Test_newPostgresProductAccessor(t *testing.T) {
+	_ = newPostgresProductAccessor(nil)
+}
+
+func TestUpdateProduct(t *testing.T) {
+	t.Parallel()
+
+	var (
+		accessor *postgresProductAccessor
+		mock     sqlmock.Sqlmock
+	)
+
+	setup := func(t *testing.T) (*gomega.GomegaWithT, *sql.DB) {
+		g := gomega.NewWithT(t)
+		db, sqlMock, err := sqlmock.New(sqlmock.QueryMatcherOption(sqlmock.QueryMatcherEqual))
+		if err != nil {
+			log.Fatal("error initializing mock:", err)
+		}
+
+		accessor = newPostgresProductAccessor(db)
+		mock = sqlMock
+
+		return g, db
+	}
+
+	t.Run("When updating product successfully, should return no error", func(t *testing.T) {
+		g, db := setup(t)
+		defer db.Close()
+
+		initialProduct := Product{
+			ID:          "inv1",
+			Name:        "Product A",
+			Description: "Initial Description",
+		}
+
+		updatedProduct := Product{
+			ID:          "inv1",
+			Name:        "Product A Updated",
+			Description: "Updated Description",
+		}
+
+		mock.ExpectExec(`UPDATE product SET name = $1 description = $2 WHERE id = $3`).
+			WithArgs(updatedProduct.Name, updatedProduct.Description, updatedProduct.ID).
+			WillReturnResult(sqlmock.NewResult(1, 1))
+
+		ctx := context.Background()
+		err := accessor.UpdateProduct(ctx, initialProduct.ID, updatedProduct)
+
+		g.Expect(err).To(gomega.BeNil())
+	})
+
+	t.Run("When updating product fails due to database error, should return error", func(t *testing.T) {
+		g, db := setup(t)
+		defer db.Close()
+
+		updatedProduct := Product{
+			ID:          "inv1",
+			Name:        "Product A Updated",
+			Description: "Updated Description",
+		}
+
+		mock.ExpectExec(`UPDATE product SET name = $1 description = $2 WHERE id = $3`).
+			WithArgs(updatedProduct.Name, updatedProduct.Description, updatedProduct.ID).
+			WillReturnError(errors.New("some database error"))
+
+		ctx := context.Background()
+		err := accessor.UpdateProduct(ctx, updatedProduct.ID, updatedProduct)
+
+		g.Expect(err).ToNot(gomega.BeNil())
+	})
+
+	t.Run("When updating product and item not found, should return error", func(t *testing.T) {
+		g, db := setup(t)
+		defer db.Close()
+
+		updatedProduct := Product{
+			ID:          "nonexistent_inv",
+			Name:        "Non-existent Product",
+			Description: "This item does not exist",
+		}
+
+		mock.ExpectExec(`UPDATE product SET name = $1 description = $2 WHERE id = $3`).
+			WithArgs(updatedProduct.Name, updatedProduct.Description, updatedProduct.ID).
+			WillReturnError(errors.New("product item not found"))
+
+		ctx := context.Background()
```

Comment:
Why do we write `ctx := context.Background()` here?

Reply:
I think we use context.Background() because we want to pass a empty root context with no values or indications. On this case it is the best because we need to pass a new empty context.